etcd/watchmanager/types.go

```diff
@@ -35,7 +35,7 @@ type WatchManager interface {
 }
 
 // UpdateFn is called when an event on the watch channel happens
-type UpdateFn func(key string) error
+type UpdateFn func(key string, events []*clientv3.Event) error
```

Comment:
You could consider adding a new UpdateFn one with events and one without? would that make things clearer?